linera-execution/src/wasm/wasmer.rs

```diff
@@ -253,17 +257,21 @@ impl CachedContractModule {
 
     /// Creates a new [`Engine`] to compile a contract bytecode.
     fn create_compilation_engine() -> Engine {
-        let mut compiler_config = Singlepass::default();
-        compiler_config.canonicalize_nans(true);
+        #[cfg(not(web))]
+        {
+            let mut compiler_config = wasmer::Singlepass::default();
+            compiler_config.canonicalize_nans(true);
+
+            wasmer::sys::EngineBuilder::new(compiler_config).into()
+        }
 
-        EngineBuilder::new(compiler_config).into()
+        #[cfg(web)]
+        wasmer::Engine::default()
     }
 
     /// Creates a [`Module`] from a compiled contract using a headless [`Engine`].
     pub fn create_execution_instance(&self) -> Result<(Engine, Module), anyhow::Error> {
-        use wasmer::NativeEngineExt;
-
-        let engine = Engine::headless();
+        let engine = Engine::default();
```

Comment:
Maybe @jvff to confirm this one?

Reply:
This one [is fine](https://docs.rs/wasmer/latest/src/wasmer/sys/engine.rs.html#115-119), but I'd love @jvff's opinion on the safety of removing `canonicalize_nans` on the Web.

Reply:
As discussed on call, this should be an improbable edge case that doesn't affect consensus, only the possibility of a client proposing a block. We should fix it eventually, but it's not a blocker.